gnd/src/main/java/com/google/android/gnd/repository/local/Edit.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.repository.local;
+
+import static java8.util.J8Arrays.stream;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.room.ColumnInfo;
+import androidx.room.TypeConverter;
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.AutoValue.CopyAnnotations;
+import org.json.JSONObject;
+
+/** Fields describing operation type, old, and new values, common to Feature and Record edits. */
+@AutoValue
+public abstract class Edit {
+  public enum Type implements IntEnum {
+    UNKNOWN(0),
+    CREATE(1),
+    UPDATE(2),
+    DELETE(3);
+
+    private final int intValue;
+
+    Type(int intValue) {
+      this.intValue = intValue;
+    }
+
+    public int intValue() {
+      return intValue;
+    }
+
+    @TypeConverter
+    public static int toInt(@Nullable Edit.Type value) {
+      return IntEnum.toInt(value, UNKNOWN);
+    }
+
+    @NonNull
+    @TypeConverter
+    public static Edit.Type fromInt(int intValue) {
+      return IntEnum.fromInt(stream(values()), intValue, UNKNOWN);
+    }
+  }
+
+  @CopyAnnotations
+  @ColumnInfo(name = "type")
+  @NonNull
+  public abstract Type getType();
+
+  @CopyAnnotations
+  @ColumnInfo(name = "old_values")
+  @Nullable
+  public abstract JSONObject getOldValues();
```

Comment:
@navinsivakumar re your comment on previous version, if we're using JSON for both data in Record and changes in Edits, it would be more consistent to let key:value pairs be stored inside the JSON in both rather than the key being stored separately in the Edit.

In that case, why not store multiple changes in a single Edit, especially since they tend to get created together from a single Save action? If we only store the modified pairs, merging is no different than previously discussed ; we'd simply treat the presence of a key:value pair in an Edit UPDATE operation as if they were individual operations. Some illustrative examples:

**User populates Record for the first time:**
*oldValues*
```json
{}

```
*newValues*
```json
{"name": "Brutus", "breed": "Yorkshire Terrier"}
```

**User updates name:**
*oldValues*
```json
{"name": "Brutus"}
```

*newValues*
```json
{"name": "Fido"}
```

Wdyt? If you agree I'll make sure to document this in Javadoc.

Reply:
Sure, that works. And (should we support it) you would express adding a kv pair by including the key in newValues but not old (and similar for delete). That works for me.